onadata/apps/logger/tests/test_tasks.py

```diff
@@ -61,3 +67,29 @@ def test_invalid_pk(self, mock_logger, mock_set_perms):
         set_entity_list_perms_async.delay(sys.maxsize)
         mock_set_perms.assert_not_called()
         mock_logger.assert_called_once()
+
+
+class UpdateProjectDateModified(TestBase):
+    """Tests for apply_project_date_modified_async"""
+
+    def setUp(self):
+        super().setUp()
+        self.project = get_user_default_project(self.user)
+
+    def test_update_project_date_modified(self):
+        """Test project date_modified field is updated"""
+        project_ids = cache.get(BATCH_PROJECT_IDS_CACHE, {})
+        project_ids[self.project.pk] = timezone.now()
+        initial_date_modified = self.project.date_modified
+        cache.set(BATCH_PROJECT_IDS_CACHE, project_ids, timeout=300)
+
+        apply_project_date_modified_async.delay()
+        self.project.refresh_from_db()
+        current_date_modified = self.project.date_modified
+
+        # check that date_modified has changed
+        self.assertNotEqual(initial_date_modified, current_date_modified)
+
+        # check if current date modified is greater than initial
+        self.assertGreater(current_date_modified, initial_date_modified)
+        cache.delete(BATCH_PROJECT_IDS_CACHE)
```

Comment:
I think we should assert that the cache is None since it is deleted after the task completes